ui/src/main/kotlin/team/duckie/quackquack/ui/button.kt

```diff
@@ -0,0 +1,539 @@
+                            painter = trailingIcon.asPainter(),
+                            colorFilter = currentColorFilter,
+                            contentScale = ContentScale.Fit,
+                        ),
+                )
+            }
+        },
+    ) { measurables, constraints ->
+        val leadingIconMeasurable = measurables.fastFirstOrNull { measurable ->
+            measurable.layoutId == LeadingIconLayoutId
+        }
+        val textMeasurable = measurables.fastFirstOrNull { measurable ->
+            measurable.layoutId == TextLayoutId
+        }!!
+        val trailingIconMeasurable = measurables.fastFirstOrNull { measurable ->
+            measurable.layoutId == TrailingIconLayoutId
+        }
+
+        val looseConstraints = constraints.asLoose(width = true, height = true)
+
+        val horizontalPadding = contentPadding.horizontal.roundToPx() * 2
+        val verticalPadding = contentPadding.vertical.roundToPx() * 2
+
+        val baselineConstraints = constraints.offset(
+            horizontal = -horizontalPadding,
+            vertical = -verticalPadding,
+        )
+        val textPlaceable = textMeasurable.measure(baselineConstraints)
+
+        val width = constraints.constrainWidth(textPlaceable.width + horizontalPadding)
+        val height = constraints.constrainHeight(textPlaceable.height + verticalPadding)
+
+        layout(width = width, height = height) {
+            val textPlaceableX = Alignment.CenterHorizontally.align(
+                size = textPlaceable.width,
+                space = width,
+                layoutDirection = layoutDirection,
+            )
+            textPlaceable.place(
+                x = textPlaceableX,
+                y = Alignment.CenterVertically.align(
+                    size = textPlaceable.height,
+                    space = height,
+                ),
+            )
+
+            if (leadingIconMeasurable != null) {
+                val leadingIconPlaceable = leadingIconMeasurable.measure(looseConstraints)
+                leadingIconPlaceable.place(
+                    x = textPlaceableX - iconSpacedBy.roundToPx() - leadingIconPlaceable.width,
+                    y = Alignment.CenterVertically.align(
+                        size = leadingIconPlaceable.height,
+                        space = height,
+                    ),
+                )
+            }
+            if (trailingIconMeasurable != null) {
+                val trailingIconPlaceable = trailingIconMeasurable.measure(looseConstraints)
+                trailingIconPlaceable.place(
+                    x = textPlaceableX + textPlaceable.width + iconSpacedBy.roundToPx(),
+                    y = Alignment.CenterVertically.align(
+                        size = trailingIconPlaceable.height,
+                        space = height,
+                    ),
+                )
+            }
+        }
+    }
+}
+
+public fun Modifier.fontScaleAwareIconSize(baseline: Dp = 24.dp): Modifier {
```

Comment:
### TODO

1. Add the `@Stable` annotation
2. Moving functions to the `ux` package
3. Apply `remember` to `defaultSize`